nova,neutron,glance: keystone auth_token memcached protection

stefannica opened the pull request:
This PR continues the work done in #1090 by configuring the auth_token keystone middleware for all services currently using memcached caching (neutron, glance, nova) to authenticate and encrypt the token data stored in memcached using a generated secret key.
    
More information: 
- https://docs.openstack.org/keystonemiddleware/latest/middlewarearchitecture.html#memcache-protection

cmurphy commented on chef/data_bags/crowbar/migrate/glance/201_add_memcache_secret_key.rb:
```diff
@@ -0,0 +1,12 @@
+end
+
+def downgrade(ta, td, a, d)
+  a.delete["memcache_secret_key"]
```
This needs to be `a.delete("memcache_secret_key")` (parentheses not brackets)

stefannica replied:
ooops !  good eye @cmurphy ! thanks !

cmurphy commented on chef/data_bags/crowbar/template-neutron.json:
```diff
@@ -172,7 +173,7 @@
       "crowbar-revision": 0,
       "crowbar-applied": false,
-      "schema-revision": 200,
+      "schema-revision": 201,
```
This needs to be bumped

stefannica commented:
HA gating job fail is fixed by #1125

stefannica commented:
@cmurphy , @sayalilunkad would you please recheck this PR ? All I did since your last review was I rebased a couple of times and updated the schema version numbers to stay on top of master.
